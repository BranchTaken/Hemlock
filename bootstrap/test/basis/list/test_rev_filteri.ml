open! Basis.Rudiments
open! Basis
open List

let test () =
  let lists = [
    [];
    [0L];
    [0L; 1L];
    [0L; 1L; 2L];
    [0L; 1L; 2L; 3L];
    [0L; 1L; 2L; 3L; 4L];
  ] in
  iter lists ~f:(fun l ->
    let f i _ = (i % 2L = 0L) in
    File.Fmt.stdout
    |> Fmt.fmt "[rev_]filteri "
    |> (pp Uns.pp) l
    |> Fmt.fmt " -> "
    |> (pp Uns.pp) (filteri l ~f)
    |> Fmt.fmt " / "
    |> (pp Uns.pp) (rev_filteri l ~f)
    |> Fmt.fmt "\n"
    |> ignore
  )

let _ = test ()
