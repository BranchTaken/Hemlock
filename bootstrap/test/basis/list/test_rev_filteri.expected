[rev_]filteri [] -> [] / []
[rev_]filteri [0] -> [0] / [0]
[rev_]filteri [0; 1] -> [0] / [0]
[rev_]filteri [0; 1; 2] -> [0; 2] / [2; 0]
[rev_]filteri [0; 1; 2; 3] -> [0; 2] / [2; 0]
[rev_]filteri [0; 1; 2; 3; 4] -> [0; 2; 4] / [4; 2; 0]
